### PR TITLE
add cspell as new spell checker

### PR DESCRIPTION
**Pull request by tuminoid**
Add cspell as spell checker to replace yaspeller that was removed earlier.

Word list from yaspeller ignore list added back as is, and the new or remaining spell checking issues fixed.

### CODE REVIEW

**Comment by tuminoid** [comment]
/hold
Needs https://github.com/metal3-io/project-infra/pull/624 to merge first.

File: hack/spellcheck.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+if [ "${IS_CONTAINER}" != "false" ]; then
+    npm install -g cspell
+    npx cspell lint --show-suggestions -c /workdir/.cspell-config.json ./**/*.md
+else
+    "${CONTAINER_RUNTIME}" run --rm \
+        --env IS_CONTAINER=TRUE \
+        --volume "${PWD}:/workdir:ro,z" \
+        --entrypoint sh \
+        --workdir /workdir \
+        docker.io/library/node:21.6.0-bookworm@sha256:0ded28778059262bd3c066b609186e5b6c89550a9362dce4309ad67c95af0d77 \
```
Comment (kashifest) [inline hack/spellcheck.sh:29]: is this the tool we are using https://github.com/streetsidesoftware/cspell-cli/pkgs/container/cspell and if it is why not we use ghcr.io/streetsidesoftware/cspell:8.3.2 instead?

Comment (kashifest) [reply hack/spellcheck.sh:29]: The image size is much small + you get a pinned version of cspell
Screenshot 2024-01-25 at 15 07 51

Comment (tuminoid) [reply hack/spellcheck.sh:29]: cspell-cli is different from cspell, but I can check if it achieves the same.

Comment (kashifest) [reply hack/spellcheck.sh:29]: yeah but here as you can see I have downloaded the cspell image not the cspell-cli, I am guessing this should have the cspell binary that we are installing in the script currently

Comment (kashifest) [reply hack/spellcheck.sh:29]: It might not have everything that we need but worth a check, thanks for checking it.

Comment (tuminoid) [reply hack/spellcheck.sh]: It works the same, so its much better this way. Thanks for the suggestion.

**Comment by tuminoid** [comment]
Updated per @kashifest to use cspell-cli image. Much smaller image to download.

**Comment by maxrantil** [comment]
LGTM

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by Rozzii — APPROVED** [review]
/lgtm

**Comment by tuminoid** [comment]
/retest
Spellcheck should trigger now.

**Comment by kashifest** [comment]
why do I see two spellcheck jobs here ?

**Comment by tuminoid** [comment]
> why do I see two spellcheck jobs here ?

sHellcheck and sPellcheck :)

**Comment by kashifest** [comment]
> > why do I see two spellcheck jobs here ?
> 
> sHellcheck and sPellcheck :)

Oh my my, I need my glasses

**Comment by tuminoid** [comment]
/unhold
Good to go.

**Pull request merged**